### PR TITLE
Update together model prompt and stop sequence

### PR DESCRIPTION
**Pull request by insop**
Update together model prompt and stop sequence
    
    - update prompt based on the model name
    - update stop sequence

*TEST*:
- the following models are tested using `intro.ipynb`
- completion model
```
model="mistralai/Mistral-7B-v0.1"
turbo = dspy.Together(model=model)
```
- instruction mode
```
model="mistralai/Mistral-7B-Instruct-v0.2"
turbo = dspy.Together(model=model)
```

### CODE REVIEW

File: dsp/modules/hf_client.py [hunk 1]
```diff
@@ -212,13 +212,19 @@ def __init__(self, model, **kwargs):
...
+            self.use_inst_template = True
+
         self.kwargs = {
             "temperature": 0.0,
             "max_tokens": 512,
             "top_p": 1,
             "top_k": 20,
             "repetition_penalty": 1,
             "n": 1,
+            "stop": ["\n\n", "---", "[/INST]"],
```
Comment (okhat) [inline dsp/modules/hf_client.py:227]: Is this the final thing we decided or did you want to try the \n\n---?

File: dsp/modules/bedrock.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```
Comment (insop) [inline dsp/modules/bedrock.py:1]: @JamesScharf, @okhat 

Adding this due to this error:

> TypeError: unsupported operand type(s) for |: 'type' and 'type'

File: dsp/modules/hf_client.py (hunk 1 above)
Comment (insop) [reply dsp/modules/hf_client.py]: Hi @okhat 

I was testing locally with ` \n\n---` yesterday.
Now I have updated the PR, PTAL.

Thank you.

**Comment by insop** [comment]
Hi @okhat 
Please let me know if you have any feedback/suggestion.
Thank you.

**Pull request merged**